src/components/Dashboard/Dashboard.jsx

```diff
@@ -4,10 +4,18 @@ import NavBar from './Header/Header';
 import MainContainer from './MainContainer/MainContainer';
 
 export default class Dashboard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoggedIn: props.isLoggedIn,
+    };
+  }
+
   render() {
+    const { isLoggedIn } = this.state;
     return (
       <div>
-        <NavBar />
+        <NavBar isLoggedIn={isLoggedIn} />
```

Comment:
Navbar is present in all views. To adhere to DRY principle, we should place Navbar component on the very top of the application, rather than import it to both MainView and Dashboard components